cpp/dolfinx/mesh/GraphBuilder.cpp

```diff
@@ -67,7 +65,7 @@ compute_local_dual_graph_keyed(
       // Get list of facet vertices
       auto& facet = facets[counter].first;
       for (int k = 0; k < N; ++k)
-        facet[k] = cell_vertices(i, facet_vertices(j, k));
+        facet[k] = cell_vertices.links(i)[facet_vertices(j, k)];
```

Comment:
I'd suggest hoisting `auto foo = cell_vertices.links(i);` outside of the `j` and `k` loops.